### ISSUE TITLE
DSE mem_read issue

### ISSUE THREAD

**Issue by nofiv** [cea-sec/miasm#1105]
Hello,

I've noticed that DSE's symbolic execution engine treats some memory-reads in a different way than the pure one which results in some very "ugly" expressions and loss of precise values (I can provide examples if they are required). I found out that the cause is [here](https://github.com/cea-sec/miasm/blob/master/miasm/analysis/dse.py#L112). Changing the line to `return super(EmulatedSymbExec, self).mem_read(expr_mem)` fixes the issue. Is this a bug? If no, what are the reasons for this behavior?

**Comment by serpilliere** [comment]
I am not sure to see exactly the problem: can you provide a little example?

**Comment by nofiv** [comment]
Lets say that instruction `SUB QWORD PTR [RSP], 0x7FBC0F01` is to be executed and state in the both engines contains:
```
...
RSP = RSP_init + 0xFFFFFFFFFFFFFFE8
@64[RSP_init + 0xFFFFFFFFFFFFFFE8] = R12_init
...
```
DSE engine's state after the execution:
```
...
RSP = RSP_init + 0xFFFFFFFFFFFFFFE8
@64[RSP_init + 0xFFFFFFFFFFFFFFE8] = @64[RSP_init + 0xFFFFFFFFFFFFFFE8] + 0xFFFFFFFF8043F0FF
...
```
Pure symbolic execution engine's state after the execution:
```
...
RSP = RSP_init + 0xFFFFFFFFFFFFFFE8
@64[RSP_init + 0xFFFFFFFFFFFFFFE8] = R12_init + 0xFFFFFFFF8043F0FF
...
```
I'd expect these 2 states to be the same. It also leads to further irregularities this way.

**Comment by serpilliere** [comment]
Huu. Seems totally legit to me.
@commial, do you concur?
( https://www.youtube.com/watch?v=i5j1wWY-qus )

**Comment by commial** [comment]
Yep, agreed.
Thanks for spotting and reporting it!

Implemented in #1108
